### ISSUE TITLE
integration of text selection in browse mode with native text selection on page in browser

### ISSUE THREAD

**Issue by tseykovets** [nvaccess/nvda#15964]
### Is your feature request related to a problem? Please describe.

Currently, NVDA users are unable to select text on a page opened in a browser. They can only select the text in NVDA's own virtual representation (browse mode).

This becomes a problem when The user needs to select the text directly on the page. For example, a site has a function for sending information about typos or getting help about content, which requires the user to select the text and press a certain command. The native text selection in a browser is also required, for example, to [share or link to quotes & text in Chrome](https://support.google.com/chrome/answer/10256233).

### Describe the solution you'd like

It seems important to provide an accessible way to select text directly on the page.

The best option would be to ensure that the text selection in browse mode is integrated with native text selection on page in browser. As a last resort, supporting the Caret Browsing mode would be an acceptable solution.

### Describe alternatives you've considered

All common browsers have the Caret Browsing accessibility feature for navigate and select text without using the mouse, for example, [Google Chrome](https://support.google.com/chrome/answer/10129654).
However, operation in this mode is not supported by NVDA (the content over which the Caret Browsing cursor moves is not announced).

By the way, JAWS and Narrator screen readers integrate text selection in their modes for web content with native text selection on the page in browser. Although when trying to select one or two words, larger pieces of text are often captured.

### Additional context

Description for Reproducible Testing:

#### Preconditions

* NVDA with default settings is running on the system.
* Any browser is running on the system.
* The [example.com](https://example.com/) page is open in the browser

#### Steps to reproduce:

1. Press NVDA+T three times quickly
2. Press NVDA+C
3. In browse mode, select the text "This domain is for use in illustrative examples in documents."
4. Press NVDA+F2
5. Press CTRL+C
6. Press NVDA+C

#### Expected behavior:

* In step 2, NVDA speaks the browser window title copied to the clipboard in step 1.
* In step 6, NVDA speaks the text from page selected in step 3 and copied to the clipboard in step 5.

#### Actual behavior:

* In step 2, NVDA speaks the browser window title copied to the clipboard in step 1.
* In step 6, NVDA still speaks the browser window title copied to the clipboard in step 1.

**Comment by mltony** [comment]
I agree this would be very helpful. Just wanted to give an example from my own experience. 
I've been recently working with developers from arxiv.org to make HTML versions of their pages to be more accessible with screenreaders. One of the features they have is the ability to select certain text with a mouse and then for sighted users there are options to either comment on selected text or report it. However for screenreader users (NVDA specifically, not sure to what degree Jaws and VoiceOver suffer from the same) they had to implement a separate mode where you can only select an entire paragraph by pressing a button, that requires a special mode that can be triggered by a keystroke, as currently there appears to be no better option for NVDA users to mark certain text.
Here are more details if anyone is interested: https://info.arxiv.org/about/accessible_HTML.html

**Comment by Emil-18** [comment]
NVDA 2024.1 will have a native selection mode for firefox. When on, the browse mode commands selects the visual text in the document as described. I wish this functionality could be extended to other browsers as well though.

**Comment by seanbudd** [comment]
We are currently blocked by work remaining in Chromium to support other browsers

**Comment by Brian1Gaff** [comment]
Maybe you could make this feature in an add on which gets updated as and when the other work is done on the browser.
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
  ----- Original Message -----

**Comment by Adriani90** [comment]
Closing as duplicate of #5354 which has to remain open until Chromium provides related work.

**Comment by Umit-Ulusoy** [comment]
@seanbudd @Brian1Gaff @mltony @tseykovets
Actually, the native selection feature can be integrated into most other browsers or applications using a mouse simulation package. By combining such a package with NVDA's browse mode (virtual representation), it’s possible to implement text selection with the mouse, mimicking native selection functionality.
Have you ever considered this approach?
If not, I’d like to enhance this feature using this or another approach.
